compiler/compile/OMRCompilation.cpp

```diff
@@ -992,6 +996,13 @@ int32_t OMR::Compilation::compile()
    if (_recompilationInfo)
       _recompilationInfo->startOfCompilation();
 
+#if defined(LINUX)
+   // Create the compile time profiler
+   TR::CompileTimeProfiler *perf = NULL;
+   if (self()->getOption(TR_CompileTimeProfiler))
+      perf = new (self()->trHeapMemory()) TR::CompileTimeProfiler(self(), "compileTimePerf");
```

Comment:
I think in order for this to work, you need to add `TR_ALLOC` inside the class definition. For example, with `TR_Dominators`:

https://github.com/eclipse/omr/blob/34b0b3e14425eeadde434dd2496098d83cb2530d/compiler/optimizer/InductionVariable.cpp#L5521

https://github.com/eclipse/omr/blob/34b0b3e14425eeadde434dd2496098d83cb2530d/compiler/optimizer/Dominators.hpp#L50-L54

Reply:
Nvm, turns out there are global definitions...

https://github.com/eclipse/omr/blob/34b0b3e14425eeadde434dd2496098d83cb2530d/compiler/env/TRMemory.hpp#L607-L617